vendor/harvey-os.org/namespace/builder.go

```diff
@@ -0,0 +1,133 @@
+package namespace
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+)
+
+// OpenFunc opens files
+type OpenFunc func(path string) (io.Reader, error)
+
+// Builder helps building plan9 namespaces
+type Builder struct {
+	dir  string
+	file File
+
+	open OpenFunc
+}
+
+func open1(path string) (io.Reader, error) { return os.Open(path) }
+
+// NewBuilder returns the default runner
+func NewBuilder() (*Builder, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+	return &Builder{
+		dir:  wd,
+		open: open1,
+	}, nil
+}
+
+func newRunner(wd string, b OpenFunc) (*Builder, error) {
+	return &Builder{
+		dir:  wd,
+		open: b,
+	}, nil
+}
+
+// Parse takes a path and parses the namespace file
+func (b *Builder) Parse(file string) error {
+	var f io.Reader
+	var err error
+	f, err = b.open(file)
+	if err != nil {
+		return err
+	}
+	b.file, err = Parse(f)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Run takes a namespace and runs commands defined in the namespace file
+func (b *Builder) buildNS(ns Namespace) error {
+	for _, c := range b.file {
+		var err error
+		args := []string{}
+		for _, arg := range c.args {
+			args = append(args, os.ExpandEnv(arg))
+		}
+
+		switch syzcall(c.syscall) {
```

Comment:
for me personally it would be a lot clearer if this case were on the strings. I kind of suspect you don't need syzcall func at all.